packages/angular/package.json

```diff
@@ -25,24 +25,39 @@
     "@sentry/types": "6.18.0",
     "@sentry/utils": "6.18.0",
     "rxjs": "^6.6.0",
-    "tslib": "^1.9.3"
+    "tslib": "^2.0.0"
   },
   "devDependencies": {
-    "@angular/common": "^10.0.3",
-    "@angular/core": "^10.0.3",
-    "@angular/router": "^10.0.3"
+    "@angular/common": "~10.2.5",
+    "@angular/core": "~10.2.5",
+    "@angular/router": "~10.2.5",
+    "@angular/compiler": "~10.2.5",
+    "@angular-devkit/build-angular": "~0.1002.4",
+    "@angular/cli": "~10.2.4",
+    "@angular/compiler-cli": "~10.2.5",
+    "@types/node": "^12.11.1",
+    "codelyzer": "^6.0.0",
+    "ng-packagr": "^10.1.0",
+    "ts-node": "~8.3.0",
+    "tslint": "~6.1.0",
+    "typescript": "~4.0.2",
+    "zone.js": "~0.10.2"
```

Comment:
Two questions here:

- Is there a reason we need to `~`-pin these rather than `^`-pinning them? (Not a problem if we need to, just curious.)

- Are all of these direct dev dependencies (as opposed to transitive ones)?

Reply:
1. The `~`-pins were set by the Angular CLI when the Angular project from which I started out was initialized. What I know is that each Angular version works with different ranges of dependency versions. An example would be that Angular 10.2 requires typescript 3.9.x or 4.0.x according to this compatibility [matrix](https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3). I assume hat the Angular CLI `~`-pins versions where only such specific ranges of versions are compatible with the used Angular version.

2. When I generated the Angular project first via the Angular CLI, it first set up a "outer" `package.json` in which these now dev dependencies were declared as dependencies. The second "inner" directory and`package.json` only contained dependencies required by the actual library and it relied on the outer one to provide all necessary dependencies for an Angular "environment" to work properly (such as the CLIs, compiler, `ng-packagr`, etc.). What I then did to flatten the project structure was to merge these `package.json`s into one which resembled our original one and I therefore put the Angular "environment" dependencies into the `devDependencies` section. This is btw similar to how the original Angular SDK's `package.json` registered (dev-)dependencies. Presumably, the listed dependencies are direct dependencies that are needed to compile an Angular project. However, I might be wrong here as my knowledge on direct vs. transitive dependencies is limited tbh.

Looking at this list though, I will remove `codelyzer` and `tslint` as I'm pretty sure we do not need them with our monorepo (since we're using `eslint`...).

Reply:
As for dependencies, the question I was asking was less "are these used?" (though good catch that the linters aren't) and more "which of these are the ones we actually use directly?" I was thinking if there were extras, they might be sub-dependencies (dependencies of our dependencies), but what you're saying now makes it make more sense where they all came from.

Keep in mind that we don't want the dev dependencies for an angular _project_ (which properly belong in our users' `package.json` files), we want the dev dependencies for an angular _library_, i.e., only the things which it takes to do our work as library developers. My advice would be to clear out that section entirely, and then use yarn to add the things we're actually using back in. The only thing I actually see us using anywhere is the `ng` CLI tool, which makes me think that if you cleared that section out and ran `yarn add -D <whatever-npm-package-provides-that-CLI-tool>`, you might have everything you need.

(It also might be worthwhile having a meta convo about how to deal with test projects and SDK changes in general. Happy to do that whenever.)

Reply:
Thanks for the idea of the bottom up approach for the dev dependencies. I did that and started with adding back `@angular/cli` (which we need for sure, as you correctly pointed out). What did not happen was that it would install other dependencies that it need as well. But by running `yarn build` I could identify easily which other packages I'd need and I continued adding those as well. At the end, it turned out that we needed most of the compilation and packaging related and Angular core/common/router packages (which makes sense, since we're using parts of these libraries in our code). 
However, I could get rid of `ts-node`, `@types/node` and `zone.js`.